src/services/db/GitFileSystemService.ts

```diff
@@ -168,6 +170,49 @@ export default class GitFileSystemService {
     )
   }
 
+  // Get filesystem stats of a file or directory
+  getFilePathStats(
+    repoName: string,
+    filePath: string
+  ): ResultAsync<fs.Stats, NotFoundError | GitFileSystemError> {
+    return ResultAsync.fromPromise(
+      fs.promises.stat(`${EFS_VOL_PATH}/${repoName}/${filePath}`),
+      (error) => {
+        if (error instanceof Error && error.message.includes("ENOENT")) {
+          return new NotFoundError("File/Directory does not exist")
+        }
+        if (error instanceof Error) {
+          return new GitFileSystemError(error.message)
+        }
+
+        logger.error(`Error when reading ${filePath}: ${error}`)
+        return new GitFileSystemError("An unknown error occurred")
+      }
+    )
+  }
+
+  // Reset the state of the local Git repository to a specific commit
+  rollback(
+    repoName: string,
+    commitSha: string
+  ): ResultAsync<true, GitFileSystemError> {
+    logger.warn(`Rolling repo ${repoName} back to ${commitSha}`)
+    return ResultAsync.fromPromise(
+      this.git
+        .cwd(`${EFS_VOL_PATH}/${repoName}`)
+        .reset(["--hard", commitSha])
+        .clean("f"),
```

Comment:
Does this clean up files recursively from directories? From the [documentation](https://git-scm.com/docs/git-clean), it seems you need to include an additional `d` flag in order to remove files more deeply nested

Reply:
Ah it doesn't, thanks! Fixed in a4f2b01579932fda6757bb4e6b89161ca97aa19e.

Reply:
Do we need to clean untracked files for sanity as well?

Reply:
Yup we should, `git clean` is being used for this purpose!